llvm/lib/Target/SPIRV/SPIRVTargetMachine.cpp

```diff
@@ -199,6 +199,7 @@ void SPIRVPassConfig::addPreLegalizeMachineIR() {
 bool SPIRVPassConfig::addLegalizeMachineIR() {
   addPass(new Legalizer());
   addPass(createSPIRVPostLegalizerPass());
+  addPass(createSPIRVEmitNonSemanticDIPass(&getTM<SPIRVTargetMachine>()));
```

Comment:
@bwlodarcz In prior discussions you've mentioned that the pass should be placed closer to the end of Backend's passes sequence. In this case you insert it between Post-Legalizer and Instruction Selection. I wonder if you intentionally is going to emit Debug Info before the Instruction Selection step, just in case.
@michalpaszkowski May I also ask your opinion about the proper place of this pass?

Reply:
Good catch - I missed it. It should be placed at the end of machine passes.

Reply:
Done.